src/integrations/prefect-kubernetes/prefect_kubernetes/worker.py

```diff
@@ -621,7 +646,121 @@ async def run(
 
             return KubernetesWorkerResult(identifier=pid, status_code=status_code)
 
-    async def teardown(self, *exc_info):
+    async def submit(
+        self,
+        flow: "Flow[..., R]",
+        parameters: dict[str, Any] | None = None,
+        job_variables: dict[str, Any] | None = None,
+    ) -> "PrefectFlowRunFuture[R]":
+        """
+        EXPERIMENTAL: The interface for this method is subject to change.
+
+        Submits a flow to run in a Kubernetes job.
+
+        Args:
+            flow: The flow to submit
+            parameters: The parameters to pass to the flow
+
+        Returns:
+            A flow run object
+        """
+        if self._runs_task_group is None:
+            raise RuntimeError("Worker not properly initialized")
+        flow_run = await self._runs_task_group.start(
+            partial(
+                self._submit_adhoc_run,
+                flow=flow,
+                parameters=parameters,
+                job_variables=job_variables,
+            ),
+        )
+        return PrefectFlowRunFuture(flow_run_id=flow_run.id)
+
+    async def _submit_adhoc_run(
+        self,
+        flow: "Flow[..., R]",
+        parameters: dict[str, Any] | None = None,
+        job_variables: dict[str, Any] | None = None,
+        task_status: anyio.abc.TaskStatus["FlowRun"] | None = None,
+    ):
+        """
+        Submits a flow run to the Kubernetes worker.
+        """
+        from prefect._experimental.bundles import (
+            convert_step_to_command,
+            create_bundle_for_flow_run,
+        )
+
+        job_variables = job_variables or {}
+
+        if TYPE_CHECKING:
+            assert self._client is not None
+        flow_run = await self._client.create_flow_run(
+            flow, parameters=parameters, state=Pending()
+        )
+        if task_status is not None:
+            task_status.started(flow_run)
+        # Avoid an API call to get the flow
+        api_flow = APIFlow(id=flow_run.flow_id, name=flow.name, labels={})
+        logger = self.get_flow_run_logger(flow_run)
+
+        bundle = create_bundle_for_flow_run(flow=flow, flow_run=flow_run)
+
+        # TODO: Replace this with reading the steps from the work pool
+        upload_step = json.loads(os.environ.get("PREFECT__BUNDLE_UPLOAD_STEP", "{}"))
+        execute_step = json.loads(os.environ.get("PREFECT__BUNDLE_EXECUTE_STEP", "{}"))
```

Comment:
i know this isn't a baked version but just to get things organized, this step should depend entirely on the `upload_step`, right? and so doesn't need to be set as an environment variable, at least not in the submitting process

Reply:
The execute step is also here because the worker uses it to generate the command that kicks off flow execution in the Kubernetes job. Only the upload step is run, but the worker also needs to know the execute step to when setting up the template for the execution environment.

Reply:
Right what I mean is, given an `upload_step`, shouldn't I be able to fully specify the `execute_step` (so that it's not necessary for the parent process to have _both_ specified)?

Reply:
It depends on how we implement the API. I was thinking of an explicit approach where the server returns both steps for a given work pool, so the worker doesn't need to know how to map between steps. The client configuring the work pool would need to know how to correctly set up those steps (i.e. the UI accepts a bucket and credentials block,  generates the steps, and saves them with the API). I'm open to other approaches though.